snarkos/ledger/mod.rs

```diff
@@ -112,13 +166,15 @@ impl<N: Network> Ledger<N> {
                 .and_then(get_peers_all);
 
             /// Returns the number of peers connected to the node.
-            async fn get_peers_count<N: Network>(peers: Peers<N>) -> Result<impl Reply, Rejection> {
-                Ok(reply::json(&peers.read().len()))
+            async fn get_peers_count<N: Network, E: Environment>(peers: Arc<Peers<N, E>>) -> Result<impl Reply, Rejection> {
+                Ok(reply::json(&peers.number_of_connected_peers().await))
             }
 
             /// Returns the peers connected to the node.
-            async fn get_peers_all<N: Network>(peers: Peers<N>) -> Result<impl Reply, Rejection> {
-                Ok(reply::json(&peers.read().keys().map(|addr| addr.ip()).collect::<Vec<IpAddr>>()))
+            async fn get_peers_all<N: Network, E: Environment>(peers: Arc<Peers<N, E>>) -> Result<impl Reply, Rejection> {
+                Ok(reply::json(
+                    &peers.connected_peers().await.iter().map(|addr| addr.ip()).collect::<Vec<IpAddr>>(),
```

Comment:
shouldn't we return the ports too? in the previous testnets there were many non-standard ones

Reply:
At the moment, we don't sanitize the listener ports (since we do not have a formal handshake atm), so the listener ports are the outbound ports of that peer - which usually only exists for that particular connection.